google-cloud-pubsublite/src/main/java/com/google/cloud/pubsublite/internal/wire/SubscriberImpl.java

```diff
@@ -115,13 +129,21 @@ protected void stop() {
 
   @Override
   public ApiFuture<Offset> seek(SeekRequest request) {
-    try (CloseableMonitor.Hold h = monitor.enter()) {
+    try (CloseableMonitor.Hold h =
+        monitor.enterWhenUninterruptibly(
+            new Monitor.Guard(monitor.monitor) {
+              @Override
+              public boolean isSatisfied() {
+                return !internalSeekInFlight || shutdown;
```

Comment:
If shutdown == true, we'll checkState-fail right below this. I think we should return an immediately-failed Future.

Reply:
I think this should already happen.  The catch block will catch the checkState failure and will return an immediateFailedFuture.